### PR TITLE
Add documentation for `jcmd` support

### PR DESCRIPTION
**Pull request by roberttoyonaga**
### Summary
This PR adds some documentation for the new `jcmd` support. This PR should not be merged until after https://github.com/oracle/graal/pull/9963 is integrated.

### CODE REVIEW

**Comment by roberttoyonaga** [comment]
Hi @olyagpl, now that https://github.com/oracle/graal/pull/9963 has been integrated, can you please have a look at these documentation updates when you have time?

**Comment by olyagpl** [comment]
Hello @roberttoyonaga! Yes, I can take from here, there is some information already available to process and add to the Native Image monitoring documentation (https://github.com/oracle/graal/pull/9232).

File: docs/reference-manual/native-image/JCmd.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+The adding `jcmd` support will have minimal impact on Native Image performance while idle.  However, the performance impact of the `jcmd` feature greatly depends on which diagnostic commands are invoked and their frequency. For example, invoking multiple garbage collections will have a much greater impact than dumping a single native memory tracking report. 
+
+
+## Limitations
+
+Currently, this feature is not available on Windows.
+
+### Further Reading
+
+- [Debugging and Diagnostics](DebuggingAndDiagnostics.md)
```
Comment (olyagpl) [inline docs/reference-manual/native-image/JCmd.md:83]: I would also add some JFR guide to this reading suggestion:
```suggestion
- [Build and Run Native Executables with JFR](guides/build-and-run-native-executable-with-jfr.md)
```

File: docs/reference-manual/native-image/JCmd.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+| VM.system_properties | Always available                     | Print native memory usage                                                                          |
+| VM.uptime            | Always available                     | Print VM uptime.                                                                                   |
+| VM.version           | Always available                     | Print JVM version information.                                                                     | 
+| help                 | Always available                     | Display help information.                                                                          |
+
+## Performance
+
+The adding `jcmd` support will have minimal impact on Native Image performance while idle.  However, the performance impact of the `jcmd` feature greatly depends on which diagnostic commands are invoked and their frequency. For example, invoking multiple garbage collections will have a much greater impact than dumping a single native memory tracking report.
```
Comment (olyagpl) [inline docs/reference-manual/native-image/JCmd.md:74]:
```suggestion
Adding `jcmd` support to Native Image has minimal impact on performance when the application is idle.
However, the performance impact varies significantly depending on the diagnostic commands used and how frequently they are invoked.
For example, triggering multiple garbage collections will have a much greater overhead than dumping a single native memory tracking report.
```

File: docs/reference-manual/native-image/JCmd.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+| VM.system_properties | Always available                     | Print native memory usage                                                                          |
+| VM.uptime            | Always available                     | Print VM uptime.                                                                                   |
+| VM.version           | Always available                     | Print JVM version information.                                                                     | 
+| help                 | Always available                     | Display help information.                                                                          |
+
+## Performance
+
+The adding `jcmd` support will have minimal impact on Native Image performance while idle.  However, the performance impact of the `jcmd` feature greatly depends on which diagnostic commands are invoked and their frequency. For example, invoking multiple garbage collections will have a much greater impact than dumping a single native memory tracking report. 
+
+
```
Comment (olyagpl) [inline docs/reference-manual/native-image/JCmd.md:76]: Double empty lines
```suggestion
```

**Review by olyagpl — CHANGES_REQUESTED** [review]

**Comment by olyagpl** [comment]
@roberttoyonaga, I left some rewording and cosmetic suggestions. Thank you for always finding time to contribute some documentation for a new feature!

File: docs/reference-manual/native-image/JCmd.md [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+## Enabling `jcmd` Support
+
+Support for `jcmd` is disabled by default and must be explicitly enabled at build time.
+ 
+To build a native executable with `jcmd` support, use the `--enable-monitoring=jcmd` option.
+```shell
+native-image --enable-monitoring=jcmd YourApplication
+```
+
```
Comment (christianhaeubl) [inline docs/reference-manual/native-image/JCmd.md:20]: Please briefly mention `jvmstat` as well (without `jvmstat` support, `jcmd` won't list the native image process if `jcmd` is executed without any arguments).

File: docs/reference-manual/native-image/JCmd.md [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+For more information about a specific command use 'help <command>'.
+
+```
+
+## Supported Diagnostic Commands
+The following key-value pairs are supported:
+
+| Name                 | Included with `--enable-monitoring=` | Description                                                                                        |
+|----------------------|--------------------------------------|----------------------------------------------------------------------------------------------------|
+| GC.heap_dump         | heapdump                             | Generate a HPROF format dump of the Java heap.                                                     |
```
Comment (christianhaeubl) [inline docs/reference-manual/native-image/JCmd.md:57]: `Compiler.dump_code_cache` is missing here (only available with Truffle).

File: docs/reference-manual/native-image/JCmd.md (hunk 2 above)
Comment (christianhaeubl) [inline docs/reference-manual/native-image/JCmd.md:74]: Please mention that `jcmd help <cmd>` prints the impact of a command.

**Review by christianhaeubl — COMMENTED** [review]
Thanks for the PR, I added a few comments.

File: docs/reference-manual/native-image/DebuggingAndDiagnostics.md [hunk 6]
```diff
@@ -15,4 +15,5 @@ Native Image provides utilities for debugging and inspecting the produced binary
  - For an overview of static analysis results, see [Static Analysis Reports](StaticAnalysisReports.md)
  - For performance analysis, see [Linux Perf Profiler Support in Native Image](PerfProfiling.md)
  - For an overall insight regarding build phases and the contents of a native executable, use [Build Reports](BuildReport.md)
- - For native memory tracking, see [Native Memory Tracking (NMT)](NMT.md)
\ No newline at end of file
+ - For native memory tracking, see [Native Memory Tracking (NMT)](NMT.md)
+ - For jcmd, see [Java Diagnostic Command](JCmd.md)
```
Comment (olyagpl) [inline docs/reference-manual/native-image/DebuggingAndDiagnostics.md:19]:
```suggestion
See the [Java Diagnostic Command documentation](JCmd.md) for instructions on using `jcmd`.
```

**Review by olyagpl — CHANGES_REQUESTED** [review]

File: docs/reference-manual/native-image/JCmd.md (hunk 4 above)
Comment (roberttoyonaga) [reply docs/reference-manual/native-image/JCmd.md:21]: Good idea! I've added a note about jvmstat

File: docs/reference-manual/native-image/JCmd.md (hunk 1 above)
Comment (roberttoyonaga) [reply docs/reference-manual/native-image/JCmd.md]: Good idea! Applied!

**Comment by roberttoyonaga** [comment]
Thank you @olyagpl  and @christianhaeubl for the reviews! I've applied your feedback suggestions.

**Comment by olyagpl** [comment]
@roberttoyonaga, could you please also squash the commits? Your PR should be in a good shape then.

File: docs/reference-manual/native-image/JCmd.md [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+native-image --enable-monitoring=jcmd,jfr,heapdump YourApplication
+```
+
+You may also want to include the `jvmstat` monitoring feature so your native executable can be discovered and listed with `jcmd -l` or `jcmd` with no arguments provided.
+```shell
+native-image --enable-monitoring=jcmd,jvmstat YourApplication
+```
+
+```shell
+jcmd -l
```
[the remark this answers is not in the document]
Comment (roberttoyonaga) [reply docs/reference-manual/native-image/JCmd.md:34]: oh yes, I'll fix that

**Comment by roberttoyonaga** [comment]
Thank you for the review @olyagpl. I've squashed the commits as well.

**Review by olyagpl — DISMISSED** [review]

File: docs/reference-manual/native-image/JCmd.md [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+| JFR.dump                 | jfr                                             | Copies contents of a JFR recording to file. Either the name or the recording id must be specified.  |
+| Thread.dump_to_file      | Always available                                | Dump threads, with stack traces, to a file in plain text or JSON format.                            |
+| Thread.print             | Always available                                | Print all threads with stacktraces.                                                                 |
+| VM.command_line          | Always available                                | Print the command line used to start this VM instance.                                              |
+| VM.native_memory         | nmt                                             | Print native memory usage.                                                                          |
+| VM.system_properties     | Always available                                | Print native memory usage                                                                           |
```
[the remark this answers is not in the document]
Comment (roberttoyonaga) [reply docs/reference-manual/native-image/JCmd.md:82]: Oh yes, good catch. I need to change the description for that one.

**Review by olyagpl — APPROVED** [review]

**Pull request merged**